src/fetchers/default.nix

```diff
@@ -25,15 +25,25 @@ rec {
   constructSource =
     {
       type,
-      sourceVersion ? null,
+      dependencyInfo ? { },
```

Comment:
Maybe instead of `dependencyInfo` we should directly pass `pname` and `version` here. That makes the interface more clear and simple I believe.

That won't allow us to raise the warning anymore, but maybe the warning should instead be triggered by `simpleTranslate` in `utils/translator.nix` instead.

Reply:
Hmm yeah, that's a good idea. I see that `constructSource` is also used in other places, so issuing the warning in `simpleTranslate` sounds better.

Reply:
Should be done in latest commit.